posthog/api/test/test_decide.py

```diff
@@ -146,6 +146,82 @@ def test_feature_flags(self):
             self.assertEqual(response.status_code, status.HTTP_200_OK)
         self.assertEqual(response.json()["featureFlags"], ["default-flag"])
 
+    def test_feature_flags_v2(self):
+        self.team.app_urls = ["https://example.com"]
+        self.team.save()
+        self.client.logout()
+        Person.objects.create(team=self.team, distinct_ids=["example_id"], properties={"email": "tim@posthog.com"})
+        FeatureFlag.objects.create(
+            team=self.team, rollout_percentage=50, name="Beta feature", key="beta-feature", created_by=self.user,
+        )
+        FeatureFlag.objects.create(
+            team=self.team,
+            filters={"groups": [{"properties": [], "rollout_percentage": None}]},
+            name="This is a feature flag with default params, no filters.",
+            key="default-flag",
+            created_by=self.user,
+        )  # Should be enabled for everyone
+        FeatureFlag.objects.create(
```

Comment:
I'm thinking we need one more test case–

A multivariate flag w/ a rollout percentage specified – If I'm reading things right, I  _think_ that's something we are supporting?

So basically a flag where:
1. The audience == users with realm == cloud
2. The rollout percentage is 80% of this audience
3. There are 4 variants: Variant A, B, C D
4. A, B, C,Ds rollout percentage sums to 100

If we are saying that for multivariate flags we won't allow audience level roll-outs and will just defer to the variant rollouts, I think that's acceptable too but I don't think the code is written in such a way given we are still checking the top level is_match first before jumping into the variants